src/Queue/SearchNoHitsProcessor.php

```diff
@@ -119,6 +128,37 @@ public function process(Message $message, Context $session)
                     'source' => $basicPid,
                 ]);
             }
+        } else {
+            // Try to search the data well and match source entity. This might work as there is an race between when
+            // vendors have a given cover and when the material is indexed into the data-well.
+            $type = $processMessage->getIdentifierType();
+            $material = $this->searchService->search($identifier, $type);
+
+            foreach ($material->getIdentifiers() as $is) {
+                $sourceRepos = $this->em->getRepository(Source::class);
+                $source = $sourceRepos->findOneBy([
+                    'matchId' => $is->getId(),
+                    'matchType' => $is->getType(),
+                ]);
```

Comment:
Theoretically we can have multiple sources for any `id`/`type` pair. We should load them all and use the one from the best ranked vendor

Reply:
Fixed